src/mscorlib/src/System/Threading/ClrThreadPoolBoundHandle.cs

```diff
@@ -77,34 +77,7 @@ public SafeHandle Handle
         /// </remarks>
         public static ThreadPoolBoundHandle BindHandle(SafeHandle handle)
         {
-            if (handle == null)
-                throw new ArgumentNullException(nameof(handle));
-
-            if (handle.IsClosed || handle.IsInvalid)
-                throw new ArgumentException(SR.Argument_InvalidHandle, nameof(handle));
-
-            try
-            {
-                // ThreadPool.BindHandle will always return true, otherwise, it throws. See the underlying FCall
-                // implementation in ThreadPoolNative::CorBindIoCompletionCallback to see the implementation.
-                bool succeeded = ThreadPool.BindHandle(handle);
```

Comment:
Should it actually be ThreadPool.BindHandle that throws the PNSE, with the rest of this method not changing?  Seems like we'd want the arg validation on handle, for example, to be consistent across platforms, even though then once you get past that on Unix you'd hit the PNSE.